### ISSUE TITLE
Powerless Pearl Duplication (using Gadomancy)

### ISSUE THREAD

**Issue by xSable** [BluSunrize/WitchingGadgets#112]
NOTICE: ALREADY POSTED IN GADOMANCY ISSUES, FIGURE IT BEST FOR BOTH TO KNOW!

So here's the deal:

This bug occurs when using Gadomancy(Gado) and Witching Gadgets(WG).

Specifically when making Primordial Armors (WG) and using Disguise Infusion (Gado). When you first craft the armors, it uses 3 primordial pearls, but when the infusion is finished it drops the 3 powerless pearls, so you can recharge them.

Now, once you've made them, let's say you want to disguise them.

You go through the infusion, and bam, 3 more powerless pearls drop. Now, just recharge them all, and you've got 6 pearls when you started with 3.

That's definitely a problem.

EDIT: According to Gado Devs, they won't be fixing it.

**Comment by the-vindex** [comment]
The bug is not with Gadomancy but with WG - if you use Primordial Armor in infusion crafting again, e.g. to add goggles of revealing, you will get additional pearls. Actually any crafting works - just do any enchant on it and use Spellbinding cloth to get even more pearls.

**Comment by xSable** [comment]
But isn't that a bug then?

**Comment by the-vindex** [comment]
Yes, and nobody will fix it, because they have moved on to MC 1.8+.

**Comment by xSable** [comment]
Good stuff. Thanks.

**Comment by Shahelion** [comment]
The initial wording confuses me as to where the bug is. The initial infusion uses and drops three pearls, and so does the second one, so where are the extra pearls coming from?

EDIT: Nevermind, I think I understand now...any Infusion that gives Primordial armor as a result also gives three pearls as additional output, regardless of actual input?

The pearls should be handled like recipes that use the contents of a bucket and give back an empty bucket, based on the input instead of based on the output.

**Comment by the-vindex** [comment]
Yes, the problem apparently is with what Thaumcraft API allows in this area - it has no support for such thing and similar behaviour is hardcoded for specific vanilla TC recipes.

**Comment by Shahelion** [comment]
I guess it never came up before because no one tried enchanting something that required a bucket of something to make...lol.

**Comment by xSable** [comment]
Yeah, the issue is that originally to create the armor, it uses the primordial pearls, and drops the empties for recharging.

Disguising the armor requires that the armor is already made, but AFTER the infusion is finished, it acts as if the armor was created again and drops the empty pearls once more.

Creation of the armor requires the pearls, and leaves behind 3 empty pearls.

Disguising the armor requires no pearls, but still leaves behind 3 empty pearls.

**Comment by KryptonCaptain** [comment]
The code Blu had in place already was so close to working, it just needed a little tweak.
So [I did](https://github.com/KryptonCaptain/WitchingGadgets/releases/tag/1.2.2). Bug fixed lads.

**Comment by xSable** [comment]
You may be a hero, will test when I have time.
